product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2015 OdooMRP team
+# Copyright 2015 AvanzOSC
+# Copyright 2015 Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import datetime
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+    _rec_names_search = [
+        "customer_ids.name",
+        "customer_ids.code",
+        "product_tmpl_id.customer_ids.name",
+        "product_tmpl_id.customer_ids.code",
+    ]
+
+    def name_get(self):
+        res = super(ProductProduct, self.with_context(customerinfo=True)).name_get()
+        return res
+
+    def _get_price_from_customerinfo(self, partner_id):
+        self.ensure_one()
+        if not partner_id:
+            return 0.0
+        partner = self.env["res.partner"].browse(partner_id)
+        customerinfo = self._select_customerinfo(partner=partner)
+        if customerinfo:
+            return customerinfo.price
+        return 0.0
+
+    def price_compute(
+        self, price_type, uom=False, currency=False, company=None, date=False
+    ):
+        if price_type == "partner":
+            partner_id = self.env.context.get(
+                "partner_id", False
+            ) or self.env.context.get("partner", False)
+            if partner_id and isinstance(partner_id, models.BaseModel):
+                partner_id = partner_id.id
+            prices = super().price_compute(
+                "list_price", uom, currency, company, date=date
+            )
+            for product in self:
+                price = product._get_price_from_customerinfo(partner_id)
+                if not price:
+                    continue
+                prices[product.id] = price
+                if not uom and self._context.get("uom"):
+                    uom = self.env["uom.uom"].browse(self._context["uom"])
+                if not currency and self._context.get("currency"):
+                    currency = self.env["res.currency"].browse(
+                        self._context["currency"]
+                    )
+                if uom:
+                    prices[product.id] = product.uom_id._compute_price(
+                        prices[product.id], uom
+                    )
+                if currency:
+                    date = self.env.context.get("date", datetime.datetime.now())
+                    prices[product.id] = product.currency_id._convert(
+                        prices[product.id], currency, company, date
+                    )
+            return prices
+        return super().price_compute(price_type, uom, currency, company, date=date)
+
+    def _prepare_domain_customerinfo(self, params):
+        self.ensure_one()
+        partner_id = params.get("partner_id")
+        return [
+            ("name", "=", partner_id),
```

Comment:
```suggestion
            ("parnter_id", "=", partner_id),
```
The name field was changed to partner_id in 16.0